x/launch/keeper/request.go

```diff
@@ -112,3 +116,81 @@ func (k Keeper) GetAllRequest(ctx sdk.Context) (list []types.Request) {
 
 	return
 }
+
+// applyRequest approves the request and performs
+// the launch information changes
+func applyRequest(
+	ctx sdk.Context,
+	k Keeper,
+	chainID string,
+	request types.Request,
+) error {
+	cdc := codectypes.NewInterfaceRegistry()
+	types.RegisterInterfaces(cdc)
+
+	var content types.RequestContent
+	if err := cdc.UnpackAny(request.Content, &content); err != nil {
+		return spnerrors.Critical(err.Error())
+	}
+
+	switch c := content.(type) {
+	case *types.GenesisAccount:
+		_, foundGenesis := k.GetGenesisAccount(ctx, chainID, c.Address)
+		_, foundVested := k.GetVestedAccount(ctx, chainID, c.Address)
+		if foundVested || foundGenesis {
```

Comment:
We should also return a critical error when `foundGenesis && foundVested`  the same as `AccountRemoval` and also `VestedAccount`

To factorize things, we could implement a local method `CheckAccountExistence(...) (bool, err)` that returns `false` is neither GenesisAccount nor VestedAccount exists, `true` if one of them exists and return the critical error if both exist